core/server/config/index.js

```diff
@@ -118,6 +118,15 @@ ConfigManager.prototype.set = function (config) {
         this._config.theme.navigation = config.theme.navigation;
     }
 
+    // Special case for theme.timezone, which should be overridden not merged
+    if (config && config.theme && config.theme.timezone) {
+        this._config.theme.timezone = config.theme.timezone;
+    } else if (this._config && this._config.theme) {
```

Comment:
With this condition you achieve having `this._config.theme.timezone` undefined.
The `ConfigManager.set` fn is called twice and that's why the logic here works by coincidence.